libs/langchain/langchain/memory/chat_message_histories/rocksetdb.py

```diff
@@ -0,0 +1,258 @@
+from datetime import datetime
+from time import sleep
+from typing import Any, Callable, List, Union
+from uuid import uuid4
+
+from langchain.schema import BaseChatMessageHistory
+from langchain.schema.messages import BaseMessage, _message_to_dict, messages_from_dict
+
+
+class RocksetChatMessageHistory(BaseChatMessageHistory):
+    """Uses Rockset to store chat messages.
+
+    To use, ensure that the `rockset` python package installed.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.memory.chat_message_histories import (
+                RocksetChatMessageHistory
+            )
+            from rockset import RocksetClient
+
+            history = RocksetChatMessageHistory(
+                session_id="MySession",
+                client=RocksetClient(),
+                collection="langchain_demo",
+                sync=True
+            )
+
+            history.add_user_message("hi!")
+            history.add_ai_message("whats up?")
+
+            print(history.messages)
+    """
+
+    # You should set these values based on your VI.
+    # These values are configured for the typical
+    # free VI. Read more about VIs here:
+    # https://rockset.com/docs/instances
+    SLEEP_INTERVAL_MS = 5
+    ADD_TIMEOUT_MS = 5000
+    CREATE_TIMEOUT_MS = 20000
+
+    def _wait_until(self, meth: Callable, timeout: int, **meth_params: Any) -> None:
```

Comment:
nit: meth not a super obvious abbreviation, maybe just method would be clearer

Reply:
Fixed in https://github.com/langchain-ai/langchain/pull/8940/commits/76a9d8c4e9cbf5818028c32450e8f88efd4338c1.